Add FQDN endpoint identifyer (#1464)
ietf-tapswg/api-drafts: draft-ietf-taps-interface.md

An Endpoint object can be configured with the following identifiers:

> - FQDN (string containing a Fully Qualified Domain Name):
> 
> ~~~
> RemoteSpecifier.FQDN("example.com")
> ~~~
> 
> - HostName (string that gets qualified while resolving, e.g., using DNS search domains):

Reviewer: ```suggestion
- HostName (a string, which SHOULD be a Fully Qualified Domain Name):
```

Note that an IPv6 address specified with a scope zone ID (e.g. `fe80::2001:db8%en0`)
is equivalent to `WithIPAddress` with an unscoped address and `WithInterface ` together.

> Applications creating Endpoint objects using `WithHostName` should consider using fully-qualified
> domain names (FQDNs) instead. Not providing an FQDN will result in the Transport Services Implementation

Reviewer: ```suggestion
Applications creating Endpoint objects using `WithHostName` SHOULD provide fully-qualified
domain names (FQDNs). Not providing an FQDN will result in the Transport Services Implementation
```
